src/traces/isosurface/calc.js

```diff
@@ -12,18 +12,33 @@ var colorscaleCalc = require('../../components/colorscale/calc');
 
 module.exports = function calc(gd, trace) {
 
-    var vMin = trace.isomin;
-    var vMax = trace.isomax;
-    if(vMin === undefined) vMin = Math.min.apply(null, trace.value);
-    if(vMax === undefined) vMax = Math.max.apply(null, trace.value);
+    function findMin(arr) {
+        var min = Infinity;
+        var len = arr.length;
+        for(var q = 0; q < len; q++) {
+            if(min > arr[q]) {
+                min = arr[q];
+            }
+        }
+        return min;
+    }
 
-    if(vMin === vMax) return;
-    if(vMin > vMax) {
-        var vTmp = vMin;
-        vMin = vMax;
-        vMax = vTmp;
+    function findMax(arr) {
```

Comment:
Can we turn `findMin` and `findMax` into one single loop like e.g. this one

https://github.com/plotly/plotly.js/blob/7d67f8011a0200cc09535cebde8de34199f85a09/src/traces/cone/calc.js#L21-L32

in `cone/calc.js`.

Reply:
Thanks. `isosurface/calc.js` is much cleaner now!